client/my-sites/email/email-non-domain-owner-message/index.tsx

```diff
@@ -0,0 +1,140 @@
+import { SiteDetails } from '@automattic/data-stores';
+import { TranslateResult, useTranslate } from 'i18n-calypso';
+import { stringify } from 'qs';
+import { useDispatch } from 'react-redux';
+import { useDomainOwnerUserName } from 'calypso/components/data/query-domain-owner-username';
+import PromoCard from 'calypso/components/promo-section/promo-card';
+import { ResponseDomain } from 'calypso/lib/domains/types';
+import { emailManagementPurchaseNewEmailAccount } from 'calypso/my-sites/email/paths';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+
+import './style.scss';
+
+type EmailNonDomainOwnerMessageProps = {
+	domain?: ResponseDomain;
+	selectedSite?: SiteDetails | null;
+	source: 'email-comparison' | 'email-management';
+	usePromoCard: boolean;
+};
+
+export const EmailNonDomainOwnerMessage = ( props: EmailNonDomainOwnerMessageProps ) => {
+	const { domain, selectedSite, source, usePromoCard } = props;
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const ownerUserName = useDomainOwnerUserName( selectedSite, domain );
+
+	const isPrivacyAvailable = domain?.privacyAvailable;
+
+	const buildQueryString = ( parameters = {} ) =>
+		parameters ? stringify( parameters, { addQueryPrefix: true, skipNulls: true } ) : '';
+
+	const buildLoginUrl = () => {
+		const redirectUrlParameter = emailManagementPurchaseNewEmailAccount(
+			selectedSite?.slug ?? '',
+			domain?.name ?? '',
+			null,
+			'login-redirect'
+		);
+
+		return `/log-in/${ buildQueryString( {
+			email_address: ownerUserName,
+			redirect_to: redirectUrlParameter,
+		} ) }`;
+	};
+
+	const contactOwnerUrl = `https://privatewho.is/${ buildQueryString( {
+		s: domain?.name,
+	} ) }`;
+
+	const loginUrl = buildLoginUrl();
+
+	const onClickLink = ( eventType: 'contact' | 'login' | 'support' ) => {
+		const properties = {
+			action: eventType,
+		};
+
+		dispatch( recordTracksEvent( `calypso_email_providers_nonowner_click`, properties ) );
+
+		if ( eventType === 'login' ) {
+			window.location.href = loginUrl;
+		}
+	};
+
+	const translateOptions = {
+		components: {
+			loginLink: <a href={ loginUrl } onClick={ () => onClickLink( 'login' ) } />,
```

Comment:
```suggestion
			loginLink: <a href={ loginUrl } onClick={ () => onClickLink( 'login' ) } rel="external" />,
```
I think it's preferable to force non-push based navigation this way instead of relying on `window.location.href = ...`. That should work better if the user `Cmd`+click's the link, for example.